### PR TITLE
[RLlib] Get rid of all these deprecation warnings.

### PR DESCRIPTION
**Pull request by gjoliver**
## Why are these changes needed?

Mostly agents -> algorithms, SampleBatch.concat_samples, and switch to use MultiAgentMixInReplayBuffer.
One significant change though is to stop MarwilConfig from printing this long warning message all the time.
This is being printed multiple times at the start of all RLlib runs even if the user is not using Marwil at all.
Changed it to print warning message only if someone is actually using MarwilConfig, and they are not setting ope_method param themselves.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ArturNiederfahrenhorst** [comment]
Even though the last commit says AlphaZero, I meant AlphaStar :)

File: rllib/agents/mock.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from ray.rllib.utils.deprecation import deprecation_warning
 
 deprecation_warning(
-    old="ray.rllib.agents.callbacks",
```
Comment (ArturNiederfahrenhorst) [inline rllib/agents/mock.py:10]: Just out of curiosity, why was this here before and why can we now change it to "mock"?

File: rllib/algorithms/marwil/marwil.py [hunk 2]
```diff
@@ -120,14 +122,7 @@ def __init__(self, algo_class=None):
             "is": {"type": ImportanceSampling},
             "wis": {"type": WeightedImportanceSampling},
         }
-        deprecation_warning(
-            old="MARWIL currently uses off_policy_estimation_methods: "
```
Comment (ArturNiederfahrenhorst) [inline rllib/algorithms/marwil/marwil.py:124]: I think @Rohan138 introduces this a couple of days ago. I may be wrong here, but if so, this would be a hard deprecation.

**Review by ArturNiederfahrenhorst — COMMENTED** [review]
We should run more tests on impala and alpha Star I believe.
The new MixIn buffer has some flaws, which are reflected in the changes that you can see in the mixin replay buffer file in my other PR: https://github.com/ray-project/ray/pull/25661/files#diff-15ba289a47d04a9e246567ba0b943478645d54c428128e261499f1b504eb9d85

File: rllib/agents/mock.py (hunk 1 above)
Comment (gjoliver) [reply rllib/agents/mock.py:10]: probably a copy & paste error.
this is the mock file, not callbacks.

**Review by ArturNiederfahrenhorst — APPROVED** [review]
